src/mongo/db/ldap/ldap_manager_impl.cpp

```diff
@@ -64,19 +64,24 @@ class LDAPManagerImpl::ConnectionPoller : public BackgroundJob {
     virtual void run() override {
         ThreadClient tc(name(), getGlobalServiceContext());
         LOG(1) << "starting " << name() << " thread";
-        stdx::unique_lock<Latch> lock{_mutex};
 
         // poller thread will handle disconnection events
         while (!_shuttingDown.load()) {
             MONGO_IDLE_THREAD_BLOCK;
-            _condvar.wait(lock, [this]{return _poll_fd >= 0 || _shuttingDown.load();});
-            if (_poll_fd < 0)
-                continue;
-            LOG(2) << "connection poller received file descriptor: " << _poll_fd;
+
             pollfd fd;
-            fd.fd = _poll_fd;
             fd.events = POLLPRI | POLLRDHUP;
             fd.revents = 0;
+
+            {
+                stdx::unique_lock<Latch> lock{_mutex};
+                _condvar.wait(lock, [this]{return _poll_fd >= 0 || _shuttingDown.load();});
+                fd.fd = _poll_fd;
+            }
+            if (fd.fd < 0)
+                continue;
+
+            LOG(2) << "connection poller received file descriptor: " << _poll_fd;
```

Comment:
need to replace _poll_fd with fd.fd in this line too since _poll_fd should be only accessed under mutex

Reply:
Replaced.